src/scripts/create-index.js

```diff
@@ -0,0 +1,129 @@
+import pkg from 'pg';
+const { Client } = pkg;
+import format from 'pg-format'
+
+/**
+ * Helper functions so we can return meaningful error messages
+ */
+
+export const canConnectToDatabase = async () => {
+    const client = new Client()
+    client.connect()
+    const sql = format('SELECT 1');
+
+    try {
+        const results = await client.query(sql);
+        if (!results.rows.length) return false
+        return true;
+    } catch (error) {
+        return false;
+    } finally {
+        client.end();
+    }
+}
+
+export const tableExists = async ({ tableName }) => {
+    const client = new Client()
+    client.connect()
+    const sql = format('SELECT EXISTS (SELECT 1 FROM information_schema.tables WHERE table_name = %L)', tableName);
+    try {
+        const results = await client.query(sql);
+        if (!results.rows.length) return false
+        return true;
+    } catch (error) {
+        return false;
+    } finally {
+        client.end();
+    }
+}
+
+/**
+ * Get text columns from a table
+ */
+
+export async function getTextColumnsFromTable({ tableName }) {
+    const client = new Client()
+    client.connect()
+    const sql = format('SELECT column_name FROM information_schema.columns WHERE table_name = %L AND data_type = %L', tableName, 'text');
+    const data = await client.query(sql);
+    client.end();
+    return data.rows.map(row => row.column_name);
+}
+
+/**
+ * Create the vector column and index
+ */
+
+export async function createColumnAndIndex({ tableName }) {
```

Comment:
I wonder if this should call `dropColumnAndIndex` first.  I'm not sure which is preferable.

Reply:
I'm not sure either, 'm leaning towards: if the command to drop is well documented then leave it as a manual action for the developer.